assets/src/design-system/components/button/index.js

```diff
@@ -0,0 +1,210 @@
+import { BUTTON_SIZES, BUTTON_TYPES, BUTTON_VARIANTS } from './constants';
+
+const Base = styled.button(
+  ({ theme }) => `
+  display: flex;
+  align-items: center;
+  justify-content: space-around;
+  box-sizing: border-box;
+  padding: 0;
+  margin: 0;
+  background: transparent;
+  border: none;
+  box-shadow: 0 0 0 2px ${theme.colors.bg.primary};
+  color: ${theme.colors.fg.primary};
+  cursor: pointer;
+  font-family: ${theme.typography.family.primary};
+  font-size: ${
+    theme.typography.presets.button[
+      THEME_CONSTANTS.TYPOGRAPHY_PRESET_SIZES.SMALL
+    ].size
+  }px;
+  font-weight: ${
+    theme.typography.presets.button[
+      THEME_CONSTANTS.TYPOGRAPHY_PRESET_SIZES.SMALL
+    ].weight
+  };
+  letter-spacing: ${
+    theme.typography.presets.button[
+      THEME_CONSTANTS.TYPOGRAPHY_PRESET_SIZES.SMALL
+    ].letterSpacing
+  }px;
+  line-height: ${
+    theme.typography.presets.button[
+      THEME_CONSTANTS.TYPOGRAPHY_PRESET_SIZES.SMALL
+    ].lineHeight
+  }px;
+  text-decoration: none;
+
+  &:disabled {
+    pointer-events: none;
+  }
+
+  &:focus {
+    box-shadow: 
+      0 0 0 2px ${theme.colors.bg.primary}, 
+      0 0 0 4px ${theme.colors.accent.secondary};
+    outline: none;
+  }
+
+  &:active {
+    outline: none;
+  }
+
+  transition: background-color 0.6s ease 0s;
+`
+);
+
+const primaryColors = ({ theme }) => css`
+  background-color: ${theme.colors.accent.primary};
+
+  &:hover,
+  &:focus,
+  &:active {
+    background-color: ${theme.colors.violet[40]};
+  }
+
+  &:disabled {
+    background-color: #efefef;
+    color: ${theme.colors.gray[20]};
+  }
+`;
+
+const secondaryColors = ({ theme }) => css`
+  background-color: #d1d1cc;
+
+  &:hover,
+  &:focus,
+  &:active {
+    background-color: ${theme.colors.gray[20]};
+  }
+
+  &:disabled {
+    background-color: #efefef;
+    color: ${theme.colors.gray[20]};
+  }
+`;
+
+const tertiaryColors = ({ theme }) => css`
+  background-color: ${theme.colors.standard.white};
+
+  &:hover,
+  &:focus,
+  &:active {
+    background-color: #efefef;
+  }
+
+  &:disabled {
+    background-color: ${theme.colors.standard.white};
+    color: ${theme.colors.gray[20]};
+  }
+`;
+
+const buttonColors = {
+  [BUTTON_TYPES.PRIMARY]: primaryColors,
+  [BUTTON_TYPES.SECONDARY]: secondaryColors,
+  [BUTTON_TYPES.TERTIARY]: tertiaryColors,
+};
+
+const ButtonRectangle = styled(Base)`
+  ${({ type }) => type && buttonColors[type]};
+  min-width: 80px;
+  height: 36px;
+  padding: 8px 16px;
+  border-radius: 4px;
+`;
+
+const ButtonCircle = styled(Base)`
+  ${({ type }) => type && buttonColors?.[type]};
+
+  ${({ size }) => `
+  width: ${size === BUTTON_SIZES.SMALL ? 32 : 56}px;
+  height:${size === BUTTON_SIZES.SMALL ? 32 : 56}px;
+  border-radius: 50%;
+
+  & > svg {
+    width: ${size === BUTTON_SIZES.SMALL ? 14 : 20}px;
+    height:${size === BUTTON_SIZES.SMALL ? 14 : 20}px;
+    margin: 0 auto;
+  }
+  `}
+
+  transition: border-radius 0.1s ease 0s;
```

Comment:
oops, this is residual from when i thought we needed to have the border radius change on active per figma. i fixed the style but not the transition. this will be removed.